lib/data_magic/document_builder.rb

```diff
@@ -90,6 +92,17 @@ def fix_field_type(type, value, key=nil)
         new_value
       end
 
+      def parse_boolean(value)
+        case value
+        when "true"
+          true
+        when "false"
+          false
+        else
+          fail InvalidDictionary, "unexpected value '#{value.inspect}' for type '#{type.inspect}' for field '#{key}'"
```

Comment:
Oh nice. I didn't notice that when I was in the code (for a few hours several Thursdays ago). Feel free to update this as necessary.